sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/Validations.scala

```diff
@@ -85,6 +95,13 @@ object Validations {
     override def failIfInvalid: Validation =
       throw InvalidComponentException(messages.mkString(", "))
 
+    override def combineWhenValid(validation: Validation): Validation = {
+      if (isValid) {
+        validation
+      } else {
+        this
+      }
+    }
```

Comment:
Since `Invalid`, we don't need to check `isValid`

```suggestion
    override def combineWhenValid(validation: Validation): Validation = this
```

Reply:
But I don't see why we are not using `++` instead. 

The goals of the validation framework is to accumulate errors. The combine is not accumulating, but ignoring the second validation. 

We should maybe not call it `combine` because that's exactly what we are not doing.

Reply:
My motivation was as follows, if I join these two validations then the root cause of the second validation error is most likely the first error, thus presenting 2 errors might overwhelm the user. In this case, Subscribe annotation on the type and on the method level is the first error, but I could also add an error that there is a missing event handler. Fixing the missing event handler might not fix the problem at all, because the correct solution would be to remove the annotation. Maybe I'm overreacting and if the solution is to remove the annotation, then the missing handler might still be the case. Yeah, I can revert it.

Reply:
I see. I think we should not care about this kind of situation and simply report all the errors we catch. 

Anyway, it seems you reverted it already, so 👍.